packages/jaeger-ui/src/model/transform-trace-data.test.js

```diff
@@ -53,3 +53,191 @@ describe('deduplicateTags()', () => {
   });
 });
+
+describe('transformTraceData()', () => {
+  const startTime = 1586160015434000;
+  const duration = 34000;
+  const traceID = 'f77950feed55c1ce91dd8e87896623a6';
+  const rootSpanID = 'd4dcb46e95b781f5';
+  const rootOperationName = 'rootOperation';
+  const serviceName = 'serviceName';
+
+  const spans = [
+    {
+      traceID,
+      spanID: '41f71485ed2593e4',
+      operationName: 'someOperationName',
+      references: [
+        {
+          refType: 'CHILD_OF',
+          traceID,
+          spanID: rootSpanID,
+        },
+      ],
+      startTime,
+      duration,
+      tags: [
+        {
+          key: 'http.url',
+          type: 'string',
+          value: 'https://example.com/getSome',
+        },
+        {
+          key: 'http.method',
+          type: 'string',
+          value: 'get',
+        },
+        {
+          key: 'http.status_code',
+          type: 'float64',
+          value: 200,
+        },
+      ],
+      processID: 'p1',
+    },
+    {
+      traceID,
+      spanID: '4f623fd33c213cba',
+      operationName: 'anotherOperationName',
+      references: [
+        {
+          refType: 'CHILD_OF',
+          traceID,
+          spanID: rootSpanID,
+        },
+      ],
+      startTime: startTime + 100,
+      duration,
+      tags: [
+        {
+          key: 'http.url',
+          type: 'string',
+          value: 'https://example.com/getSomeAnother',
+        },
+        {
+          key: 'http.method',
+          type: 'string',
+          value: 'get',
+        },
+        {
+          key: 'error',
+          type: 'bool',
+          value: true,
+        },
+        {
+          key: 'http.status_code',
+          type: 'float64',
+          value: 404,
+        },
+      ],
+      processID: 'p1',
+    },
+  ];
+
+  const rootSpanWithMissingRef = {
+    traceID,
+    spanID: rootSpanID,
+    operationName: rootOperationName,
+    references: [
+      {
+        refType: 'CHILD_OF',
+        traceID,
+        spanID: 'missingSpanId',
+      },
+    ],
+    startTime: startTime + 50,
+    duration,
+    tags: [
+      {
+        key: 'http.url',
+        type: 'string',
+        value: 'https://example.com/getRoot',
+      },
+      {
+        key: 'http.method',
+        type: 'string',
+        value: 'get',
+      },
+      {
+        key: 'http.status_code',
+        type: 'float64',
+        value: 200,
+      },
+    ],
+    processID: 'p1',
+  };
+
+  const rootSpanWithoutRefs = {
+    traceID,
+    spanID: rootSpanID,
+    operationName: rootOperationName,
+    startTime: startTime + 50,
+    duration,
+    tags: [
+      {
+        key: 'http.url',
+        type: 'string',
+        value: 'https://example.com/getRoot',
+      },
+      {
+        key: 'http.method',
+        type: 'string',
+        value: 'get',
+      },
+      {
+        key: 'http.status_code',
+        type: 'float64',
+        value: 200,
+      },
```

Comment:
Nit: are these tags relevant to the functionality being tested? They don't look like it. Could you please trim the fixtures to minimally required data?

Reply:
Sounds reasonable, thanks :)
[Removed](https://github.com/jaegertracing/jaeger-ui/pull/544/commits/246db5249943e2e0bccfff64bdff967c78dc3e0a) unnecessary tags.